servers/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -37,37 +128,198 @@
+   * @return the worker data service bind host (used by unit test only)
+   */
+  public String getDataBindHost() {
+    return mBlockWorker.getDataBindHost();
+  }
+
+  /**
+   * @return the worker data service port (used by unit test only)
+   */
+  public int getDataLocalPort() {
+    return mBlockWorker.getDataLocalPort();
+  }
 
-    System.exit(0);
+  /**
+   * @return the worker web service bind host (used by unit test only)
+   */
+  public String getWebBindHost() {
+    return mWebServer.getBindHost();
+  }
+
+  /**
+   * @return the worker web service port (used by unit test only)
+   */
+  public int getWebLocalPort() {
+    return mWebServer.getLocalPort();
+  }
+
+  /**
+   * @return the worker service handler (used by unit test only)
+   */
+  public BlockWorkerClientServiceHandler getBlockWorkerServiceHandler() {
+    return mBlockWorker.getWorkerServiceHandler();
+  }
+
+  /**
+   * Gets this worker's {@link tachyon.thrift.WorkerNetAddress}, which is the worker's hostname, rpc
+   * server port, data server port, and web server port.
+   *
+   * @return the worker's net address
+   */
+  public WorkerNetAddress getWorkerNetAddress() {
+    return mWorkerNetAddress;
+  }
+
+  /**
+   * Starts the Tachyon worker server.
+   */
+  public void start() throws Exception {
+    // NOTE: the order to start different services is sensitive. If you change it, do it cautiously.
+
+    // Start serving metrics system, this will not block
+    mWorkerMetricsSystem.start();
+
+    // Start serving the web server, this will not block
+    // Requirement: metrics system started so we could add the metrics servlet to the web server
+    // Consequence: when starting webserver, the webport will be updated.
+    mWebServer.addHandler(mWorkerMetricsSystem.getServletHandler());
+    mWebServer.startWebServer();
+
+    // Set updated net address for this worker in context
+    // Requirement: RPC, web, and dataserver ports are updated
+    // Consequence: create a NetAddress object and set it into WorkerContext
+    mWorkerNetAddress = new WorkerNetAddress(
+        NetworkAddressUtils.getConnectHost(ServiceType.WORKER_RPC, mTachyonConf),
+        mTachyonConf.getInt(Constants.WORKER_RPC_PORT), getDataLocalPort(),
+        mTachyonConf.getInt(Constants.WORKER_WEB_PORT));
+    WorkerContext.setWorkerNetAddress(mWorkerNetAddress);
+
+    // Start each worker
+    // Requirement: NetAddress set in WorkerContext, so block worker can initialize BlockMasterSync
+    // Consequence: worker id is granted
+    startWorkers();
+    LOG.info("Started worker with id {}", WorkerIdRegistry.getWorkerId());
+
+    mIsServingRPC = true;
+
+    // Start serving RPC, this will block
+    LOG.info("Tachyon Worker version {} started @ {}", Version.VERSION, mWorkerAddress);
+    mThriftServer.serve();
+    LOG.info("Tachyon Worker version {} ended @ {}", Version.VERSION, mWorkerAddress);
+  }
+
+  /**
+   * Stops the Tachyon worker server. Should only be called by tests.
```

Comment:
The comment on "Should only be called by tests." is obsolete now as we also call this method in main. Removed it from javadoc